### PR TITLE
dive list: on reload update filter status

### PR DESCRIPTION
**Pull request by bstoeger**
At some time, when introducing the global reset signal the filter
stopped being reloaded when loading a new log. This leads to very
strange UI behavior: dives disappear when editing fields unrelated
to the filter.

Therefore, when reloading the model, reset the filter. One might
argue whether this is the correct place. On the other hand, we
might even make the filter a sub-object of the dive-list model.
Let's think about this.

Partially solves #2961

Signed-off-by: Berthold Stoeger <bstoeger@mail.tuwien.ac.at>

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

The filter-status of individual dives was not reset when loading a file, leading to very strange UI behavior.

### Changes made:

1) Explicitly apply filter when reloading dive list.

### Related issues:

See discussion in #2961.
### Release note:

Done.
### Documentation change:

No.
### Mentions:

@fmatray @dirkhh

### CODE REVIEW

**Comment by bstoeger** [comment]
Added a second commit, which (hopefully) fixes a severe bug: Make a dive invisible by editing it - a new dive is made `current_dive` and thus shown. However, it is not selected and therefore edit-actions are ignored.

This could explain instances of users reporting that edits were lost!

I reckon this should go into current release if possible, though it certainly needs careful review / testing.

**Review by dirkhh — APPROVED** [review]
I agree that this might want to be a child object of the dive list.
The place where you reset the filter is odd (but then, the whole Tree/List thing is odd). What about mobile?
I played with it and it seems to do the right thing - and I definitely think it should be in 4.9.7

**Comment by dirkhh** [comment]
I want to move the CHANGELOG entries straight into the ReleaseNotes (and maybe rephrase the second one slightly), so I'll merge locally...
